tb_plugin/torch_tb_profiler/profiler/overall_parser.py

```diff
@@ -274,6 +332,139 @@ def update_steps_consider_device_side(self, runtime_node_list, device_node_list)
                     self.steps = self.steps[:keep_steps]
                     self.steps_names = self.steps_names[:keep_steps]
 
+    def calculate_gpu_utilization(self):
```

Comment:
It seems the calculate_approximated_sm_efficency and calculate_gpu_utilization only depends on the self.steps? if so, we can extract them into a separated class like GpuParser include all these logic with accept the steps as parameters to decouple the logic with current the huge file?

Reply:
Good idea. Done. Thanks!